app/src/main/java/com/marvel/marvel/budget/model/BudgetModelImpl.kt

```diff
@@ -0,0 +1,41 @@
+package com.marvel.marvel.budget.model
+
+
+import com.marvel.marvel.budget.model.resolutionstrategy.Strategy
+import com.marvel.marvel.viewmodel.ComicsViewModel
+import rx.Observable
+import rx.Observer
+import rx.Scheduler
+import rx.Subscription
+
+class BudgetModelImpl(
+    override val comics: List<ComicsViewModel>,
+    private val strategy: Strategy,
+    private val workerThread: Scheduler,
+    private val mainThread: Scheduler
+) : BudgetModel {
+
+    private var subscription: Subscription? = null
+    private var observer: Observer<List<ComicsViewModel>>? = null
+
+    override fun sortComicsListByPrice(budget: String?) {
+        val observable = Observable.fromCallable({
+            strategy.sortComicsByPricePerPage(comics, budget)
+        })
+            .subscribeOn(workerThread)
+            .observeOn(mainThread)
+            .cache()
+        observer?.let { subscription = observable.subscribe(observer) }
+    }
+
+    override fun subscribe(observer: Observer<List<ComicsViewModel>>) {
+        this.observer = observer
+    }
+
+    override fun unsubscribe() {
+        if (subscription?.isUnsubscribed == false) {
+            subscription?.unsubscribe()
+        }
```

Comment:
This can be tidier:

```
 override fun unsubscribe() {
        subscription?.let {
            if (!it.isUnsubscribed) {
                it.unsubscribe()
            }
        }
    }
```

But even cleaner still if you use a `CompositeSubscription`:
```
private val subscription = CompositeSubscription()

observer?.let { subscription.add(observable.subscribe(observer)) }

override fun unsubscribe() {
        subscription.clear()
    }
```

Advantages: 
- Uses a `val` over `var`
- Simple unsubscribe logic
- What happens if you currently call `sortComicsListByPrice` more than once? You fail to unsub everything but the last.